_docs-sources/iac/usage/using-the-library.md

```diff
@@ -0,0 +1,396 @@
+touch gw_module_guide/example/<YOUR REGION>/example/serverless-api/terragrunt.hcl
+
+mkdir -p gw_module_guide/example/<YOUR REGION>/example/serverless-api/src
+touch gw_module_guide/example/<YOUR REGION>/example/serverless-api/src/main.py
+```
+
+</TabItem>
+</Tabs>
+
+### Create the reference to the Gruntwork module
+
+Next, create the reference to the `terraform-aws-lambda` module...
+
+<Tabs groupId="tool-choice">
+<TabItem value="Terraform" label="Terraform" default>
+
+Define a module block that uses the git url `terraform-aws-lambda` module for the `source` attribute.
+
+One of the benefits of referencing modules this way is that ability to set defaults for your organization. As an example — the `terraform-aws-lambda` exposes many variables but in the module block below, we are hard coding the value `run_in_vpc` to be `false`. This will ensure that anyone consuming this module will only create AWS Lambda functions that are not in a VPC. For a full list of configuration options for this module, refer to the [Library Reference](../../reference/modules/terraform-aws-lambda/lambda/#reference).
+
+```hcl title=gw_module_guide/serverless-api/lambda/main.tf
+module "lambda" {
+  source = "git::git@github.com:gruntwork-io/terraform-aws-lambda.git//modules/lambda?ref=v0.21.9"
+
+  name        = var.name
+  runtime     = var.runtime
+  source_path = var.source_path
+  handler     = var.handler
+  run_in_vpc  = false
+  timeout     = 30
+  memory_size = 128
+}
+```
+
+Next, add the variables to the `variables.tf` file.
+
+```hcl title=gw_module_guide/serverless-api/lambda/variables.tf
+variable "name" {
+  type        = string
+  description = "Name that will be used for the AWS Lambda function"
+}
+
+variable "runtime" {
+  type        = string
+  description = "The runtime of the Lambda. Options include go, python, ruby, etc."
+}
+
+variable "source_path" {
+  type        = string
+  description = "The path to the directory containing the source to be deployed to lambda"
+}
+
+variable "handler" {
+  type        = string
+  description = "The name of the handler function that will be called as the entrypoint of the lambda"
+}
+
+```
+
+</TabItem>
+<TabItem value="Terragrunt" label="Terragrunt" default>
+
+Define the module in `_envcommon` that references the `terraform-aws-lambda`. For `terragrunt`, the source is defined in the `source` attribute of the `terraform` block.
+
+One of the benefits of referencing modules this way is that ability to set defaults for your organization. As an example — the `terraform-aws-lambda` exposes many variables but in the module block below, we are hard coding the value `run_in_vpc` to be `false`. This will ensure that anyone consuming this module will only create AWS Lambda functions that are not in a VPC. For a full list of configuration options for this module, refer to the [Library Reference](../../reference/modules/terraform-aws-lambda/lambda/#reference).
```

Comment:
```suggestion
One of the benefits of referencing modules this way is the ability to set defaults for your organization. As an example — the `terraform-aws-lambda` exposes many variables but in the module block below, we are hard coding the value `run_in_vpc` to be `false`. This will ensure that anyone consuming this module will only create AWS Lambda functions that are not in a VPC. For a full list of configuration options for this module, refer to the [Library Reference](../../reference/modules/terraform-aws-lambda/lambda/#reference).
```